src/irmin/irmin.mli

```diff
@@ -2726,43 +2726,46 @@ let () = Lwt_main.run (test ())
     We will demonstrate the use of custom merge operators by
     defining mergeable debug log files. We first define a log entry
     as a pair of a timestamp and a message, using the combinator
-    exposed by {{:https://github.com/mirage/mirage-tc}mirage-tc}:
+    exposed by {!Irmin.Type}:
 
 {[
-  module Entry = sig
-    include Irmin.Contents.Conv
-    val v: string -> t
-    val compare: t -> t -> int
-    val timestamp: t -> int
-  end = struct
-
-    type t = { timestamp: int; message : string; }
-
-    let compare x y = compare x.timestamp y.timestamp
-
-    let v message =
-      incr time;
-      { timestamp = !time; message }
-
-    let t =
-      let open Irmin.Type in
-      record "entry" (fun timestamp message -> { timestamp; message })
-      |+ field "timestamp" int    (fun t -> t.timestamp)
-      |+ field "message"   string (fun t -> t.message)
-      |> sealr
-
-    let timestamp t = t.timestamp
-
-    let pp ppf { timestamp; message } =
-      Fmt.pf ppf  "%04d: %s\n" timestamp message
-
-    let of_string str =
-      match String.cut ~sep:": " str with
-      | None -> Error (`Msg ("invalid entry: " ^ str))
-      | Some (x, message) ->
-        try Ok { timestamp = int_of_string x; message }
-        with Failure e -> Error (`Msg e)
-  end
+module Entry : sig
+  include Irmin.Contents.Conv
+  val v: string -> t
+  val compare: t -> t -> int
+  val timestamp: t -> int
+end = struct
+
+  type t = { timestamp: int; message : string; }
+
+  let compare x y = compare x.timestamp y.timestamp
+
+  let time = ref 0
+
+  let v message =
+    incr time;
+    { timestamp = !time; message }
+
+  let t =
+    let open Irmin.Type in
+    record "entry" (fun t32 message -> { timestamp = Int32.to_int t32; message })
```

Comment:
I'm not certain it's necessary, but I'm also not familiar with irmin yet -- just offering what I can while I work through the library.

WRT using `int32` and `int` int the example:

I switched to the `int32` combinator because the `int` combinator [was removed](https://github.com/mirage/irmin/commit/dbad61bf215969f6698c85a013061b2818bb1a6d). I preserved most `int` usage for convenience's sake (e.g. being able to use [`incr` in `v`](https://github.com/mirage/irmin/blob/a788144631fac5a45321f0e850860a8edcff6ad5/src/irmin/irmin.mli#L2744)).

If I'm off-base, I'll happily update the patch.

Reply:
Ha yes, I completely forgot about the removal of the `int` combinator. Thanks for the reminders :-) So your changes make perfect sense.